### PR TITLE
Update Developer Guide and PPP

### PR DESCRIPTION
**Pull request by moreTriangles**
Update introduction paragraph, export and import command, relevant plantUML diagrams and PPP draft

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -293,20 +293,89 @@ The `Github` class is first integrated into the `Person` class and added as a ne
...
+   * `ImportCommand#exportAddressBookToCsv(ReadOnlyAddressBook)`
+
+Firstly, the user command is handed over to `ExportCommandParser`. `ExportCommandParser` implements `Parser<ExportCommand>`. It implements the `parse()` method, which parses the user's command and returns an `ExportCommand` with the filename. In `parse()`, it verifies 3 requirements:
+* Arguments are not empty
+* Not more than 1 argument
+* Filename provided is a .csv or .json file
+
+The Sequence Diagram below illustrates the interactions within the `Logic` component during the first step of executing `execute("export newfriends.json")`.
+
+![ExportSequenceDiagram](images/ExportSequenceDiagramStep1.png)
```
Comment (Jai2501) [inline docs/DeveloperGuide.md:311]: Well Explained.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -293,20 +293,89 @@ The `Github` class is first integrated into the `Person` class and added as a ne
...
+Executing an Import command can be summarized with the following steps:
+1. Similar to Export, the first step is parsing the user command, which is done in `ImportCommandParser`. After verification, an `ImportCommand` is generated.
+2. Upon executing the `ImportCommand`, one of the 2 methods are called to obtain a `List<Person>` from the specified file.
+   * `ImportCommand#getImportedPersonsFromJson()`
+   * `ImportCommand#getImportedPersonsFromCsv()`
+3. By iterating through the list, each person is added to the current contact list in `Model`, skipping duplicates.
+4. Command Result with success message is returned.
+
+The sequence diagram for the first step is similar to the Export Command. The following sequence diagram illustrates the execution of an import command from part 2 onwards.
+![ImportSequenceDiagram](images/ImportSequenceDiagram.png)
```
Comment (Jai2501) [inline docs/DeveloperGuide.md:373]: Perhaps add a line break here?

**Review by Jai2501 — CHANGES_REQUESTED** [review]
Just a minor change, then would be good to merge.

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -293,20 +293,89 @@ The `Github` class is first integrated into the `Person` class and added as a ne
...
+2. Upon executing the `ExportCommand`, one of the 2 methods are called, depending on the file type specified.
+   * `ImportCommand#exportAddressBookToJson(ReadOnlyAddressBook)`
+   * `ImportCommand#exportAddressBookToCsv(ReadOnlyAddressBook)`
+
+Firstly, the user command is handed over to `ExportCommandParser`. `ExportCommandParser` implements `Parser<ExportCommand>`. It implements the `parse()` method, which parses the user's command and returns an `ExportCommand` with the filename. In `parse()`, it verifies 3 requirements:
+* Arguments are not empty
+* Not more than 1 argument
+* Filename provided is a .csv or .json file
+
+The Sequence Diagram below illustrates the interactions within the `Logic` component during the first step of executing `execute("export newfriends.json")`.
```
Comment (CheongYeeMing) [inline docs/DeveloperGuide.md:309]: The file name should be in lower camel case notation.

**Review by CheongYeeMing — APPROVED** [review]
LGTM! 👍

**Comment by moreTriangles** [comment]
Added line break and changed filename in example to `cs2103t.json` for better context

**Review by Jai2501 — APPROVED** [review]
Necessary Changes Made! LGTM!

**Pull request merged**